src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.jsx

```diff
@@ -171,22 +179,47 @@ class ProjectFiltersSettings extends AsyncComponent {
     onBlur(subfilters, e);
   };
 
-  renderDisabledFeature(fields) {
-    let {project, organization} = this.props;
-    return this.state.hooksDisabled.map(hook =>
-      hook(organization, project, null, fields)
-    );
-  }
-
   renderBody() {
-    let {features, params} = this.props;
+    let {organization, features, params} = this.props;
     let {orgId, projectId} = params;
     let {project} = this.state;
 
     if (!project) return null;
     let projectEndpoint = `/projects/${orgId}/${projectId}/`;
     let filtersEndpoint = `${projectEndpoint}filters/`;
 
+    let disabledAdvFilters = p => (
+      <FeatureDisabled
+        feature={p.features[0]}
+        alert={PanelAlert}
+        name={t('Custom Inbound Filters')}
+        message={t(
+          'Release and Error Message filtering are not enabled on your Sentry installation'
+        )}
+      />
+    );
+
+    let advCustomFilters = (
+      <Feature
+        features={['projects:custom-inbound-filters']}
+        renderDisabled={({children, ...props}) =>
+          children({...props, renderDisabled: disabledAdvFilters})}
```

Comment:
This still makes me think a bit every time I see it - I wonder if it'd be easier to understand if we extracted this logic into a prop that says something along the lines of "render only disabled component" or "continue rendering children"

Reply:
Would the hook still replace the entire children of the feature and use the children rendering logic? From my POV keeping the `renderDisabled` and disabled-hook hook rendering consistent is more clear. If we added another prop to encapsulate this logic it might become more confusing what's happening in the hook override.